### PR TITLE
Allow loading of multiple messages

### PR DESCRIPTION
**Pull request by DeepDiver1975**
@ChristophWurst 

Syntax is as follows:

GET 'http://localhost:8080/index.php/apps/mail/accounts/4/folders/SU5CT1g%3D/messages?ids=123,124'

### CODE REVIEW

**Comment by ChristophWurst** [comment]
@DeepDiver1975 review my changes please :)

File: lib/controller/messagescontroller.php [hunk 1]
```diff
@@ -356,4 +349,48 @@ private function buildHtmlBodyUrl($accountId, $folderId, $messageId) {
 		return \OC::$server->getURLGenerator()->getAbsoluteURL($htmlBodyUrl);
 	}
 
+	private function loadMultiple($accountId, $folderId, $ids) {
+		$mailBox = $this->getFolder($accountId, $folderId);
+		$account = $this->getAccount($accountId);
+		$messages = array_map(function($id) use ($accountId, $folderId, $account, $mailBox){
+			try {
+				$m = $mailBox->getMessage($id);
```
Comment (DeepDiver1975) [inline lib/controller/messagescontroller.php]: I should implement a getMessages($ids) instead of looping ....

**Comment by jancborchardt** [comment]
@DeepDiver1975 @ChristophWurst so what’s the status? Merge as is and improve later, or fix and review again? :)

**Comment by ChristophWurst** [comment]
> @DeepDiver1975 @ChristophWurst so what’s the status? Merge as is and improve later, or fix and review again? :)

If there aren't any issues, I'd say we can merge this ;-)
@jancborchardt did you test this PR?

**Comment by jancborchardt** [comment]
I don’t really know what the usecase is – seems you (@DeepDiver1975 & @ChristophWurst talked about this). So if both of you tested and it works, we can merge.

**Comment by ChristophWurst** [comment]
@DeepDiver1975 could you have a look at this? Can we merge it or is there anything left to fix/improve?

**Comment by Gomez** [comment]
Just tested basic stuff with this pr. Looks good. Folder loading seems faster...?

:+1:

**Comment by ChristophWurst** [comment]
> Folder loading seems faster...?

Shoud be faster, yes. You'll notice the biggest speedup when you clear your local storage and reload the app. As we have background fetching of messages, this can take some time depending on your network connection and if server side caching is enabled.

**Pull request merged**